src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -125,7 +125,8 @@ public Task StartAsync<TContext>(IHttpApplication<TContext> application, Cancell
 
             Contract.Assert(application != null);
 
-            _application = new ApplicationWrapper<TContext>(application);
+            _application = new ApplicationWrapper<TContext>(this, application);
+            _processRequest = new Action<RequestInitalizationContext>((ctx) => _ = _application.ProcessRequestAsync(ctx));
```

Comment:
```suggestion
            _processRequest = (ctx) => _ = _application.ProcessRequestAsync(ctx);
```

Does the above not work?

Reply:
Changed to implement IThreadPoolWorkItem